### PR TITLE
Some improvements for Posts Screen and Create Post Screen

### PR DESCRIPTION
**Pull request by SavinovJr**
1. Fixed automatic height for posts screen
2. Added placeholder for textview
3. Updated podfile

### CODE REVIEW

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift [hunk 1]
```diff
@@ -8,39 +8,93 @@
...
+            vc.crossImageView.isUserInteractionEnabled = true
+            vc.textView.placeholder = "Add post message..."
+        }
+    }
+}
+
+extension CreatePostViewController: UINavigationControllerDelegate, UIImagePickerControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        picker.dismiss(animated: true, completion: nil)
+        guard let image = info[UIImagePickerController.InfoKey.originalImage] as? UIImage else {
```
Comment (bonyadmitr) [inline dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: need to use .current and if there is no current - get original

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift [hunk 2]
```diff
@@ -8,39 +8,93 @@
...
+
+        let crossTapGesture = UITapGestureRecognizer(target: self, action: #selector(crossImageClicked))
+        crossImageView.addGestureRecognizer(crossTapGesture)
+    }
+
+    @objc func postImageClicked() {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.delegate = self
+        imagePickerController.sourceType = .photoLibrary
+        present(imagePickerController, animated: true, completion: nil)
```
Comment (bonyadmitr) [inline dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift:50]: needs to update `present` for iPad

File: dkvk-ios/DKVK/Classes/Manager/PostManager.swift [hunk 3]
```diff
@@ -10,35 +10,35 @@ import Foundation
...
 	
 	private override init() {}
 	
 	static let shared = PostManager()
 	
-	func createPost(from user: User, with text: String, completion: @escaping ItemClosure<FirebaseResult>) {
-		let postID = UUID().uuidString
-		let post = Post(text: text)
+    func createPost(from user: User, with text: String? = nil, image: UIImage? = nil, completion: @escaping ItemClosure<CreatedPostResult>) {
+        if let text = text, text.isEmpty && image == nil {
```
Comment (bonyadmitr) [inline dkvk-ios/DKVK/Classes/Manager/PostManager.swift]: `&&` can be used `,`

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift (hunk 1 above)
Comment (SavinovJr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: `UIImagePickerController.InfoKey` doesn't contain `.current` property. What do you mean exactly?

File: dkvk-ios/DKVK/Classes/Manager/PostManager.swift (hunk 3 above)
Comment (SavinovJr) [reply dkvk-ios/DKVK/Classes/Manager/PostManager.swift]: Agree, will fix

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift (hunk 2 above)
Comment (SavinovJr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift:50]: What is the problem with `present` method for `iPad`?

**Comment by bonyadmitr** [comment]
@SavinovJr,  great work!

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift (hunk 1 above)
Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: Sorry. `.original` is for Photos framework. There is `. editedImage `
https://developer.apple.com/documentation/uikit/uiimagepickercontroller/infokey/1619134-editedimage

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift (hunk 2 above)
Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift:50]: I don't remember. Maybe it needs `popoverPresentationController.sourceView` like for the alerts.

Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift:50]: Sorry, it was for the old iOS. It don't need now

File: dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift (hunk 1 above)
Comment (SavinovJr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: I added that support but to be honest when I edited image from album just adding brightness and color that `.editedImage` is still `nil`

Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: It is very interesting.
I have checked it now.
Yeap. it is `nil` for `.photoLibrary` and `.isEditing = false`.
I thought it is working like Photos framework but i was wrong.
Thank you!

Any way. We can keep new logic if we will need camera + edit mode :-)

Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: by the way can be written in short form: `info[.editedImage]`

Comment (SavinovJr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: If add `imagePickerController.allowsEditing = true` then `info[.editedImage]` will be available

Comment (bonyadmitr) [reply dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift]: > If add `imagePickerController.allowsEditing = true` then `info[.editedImage]` will be available

As I understood, it is right only for camera type, not for photoLibrary.

**Review by Hadevs — APPROVED** [review]

**Pull request merged**